pkg/ytconfig/common.go

```diff
@@ -51,6 +51,8 @@ type ClusterConnection struct {
 type AddressResolver struct {
 	EnableIPv4 bool `yson:"enable_ipv4"`
 	EnableIPv6 bool `yson:"enable_ipv6"`
+	KeepSocket bool `yson:"keep_socket"`
+	ForceTcp   bool `yson:"force_tcp"`
```

Comment:
I suggest (despite existing `EnableIP*` fields) we go with `*bool` + `omitempty` (and in the spec struct also) to avoid hardcoding ytsaurus defaults in the operator, just giving the users an option to override the defaults. 
That way it is expected we don't have a diff in our canondata (and add extra test where we are setting new fields). Also users won't have to do full update with a new release of the operator.

Reply:
How to do it in crds?
- https://github.com/ytsaurus/ytsaurus-k8s-operator/pull/324/files#diff-d4dd7e778063446c32712d7adbff415c715fe87ce7607b3762da91d51f8fe9acR10038-R10040
- https://github.com/ytsaurus/ytsaurus-k8s-operator/pull/324/files#diff-d4dd7e778063446c32712d7adbff415c715fe87ce7607b3762da91d51f8fe9acR12512-R12514

Reply:
I haven't understood your question. You use pointer in go struct and crds are generated automatically with `make generate`.

Reply:
```
`make manifests` generate:
              forceTcp:
                default: false
                type: boolean
```
And the code (operator) will then generate the full config with force_tcp (even if i not define in my .values).
But not for tests.

Aka,
- "That way it is expected we don't have a diff in our canondata" - Yes
- "Also users won't have to do full update with a new release of the operator" - No, they will have a full update :(

Reply:
ah, i think - `+kubebuilder:default:=None`  - is my option

Reply:
> "Also users won't have to do full update with a new release of the operator" - No, they will have a full update :( (cause crd's defaults will generate fullspec with force_tcp and keep_socket default values)

For users it doesn't matter if crd has new fields, I'm saying that operator will recreate all the components if it sees any diff in components static config. But if [here]("Also users won't have to do full update with a new release of the operator" - No, they will have a full update :( (cause crd's defaults will generate fullspec with force_tcp and keep_socket default values)) you use pointers with `omitempty` there will be no chagens in static configs and operator wouldn't bother users with recreating their yt cluster without need.

Reply:
> ah, i think - +kubebuilder:default:=None - is my option, sorry, forgot to edit it

Not quite sure with this defaults, maybe we can just not add `kubebuilder:default` comment here and it will be fine

Reply:
> maybe we can just not add kubebuilder:default comment here and it will be fine

Yes, this solve my questions above.
Sent a new commit.